homeassistant/components/nws/sensor.py

```diff
@@ -137,6 +139,11 @@ class NWSSensorEntityDescription(SensorEntityDescription):
         native_unit_of_measurement=UnitOfLength.METERS,
         unit_convert=UnitOfLength.MILES,
     ),
+    NWSSensorEntityDescription(
+        key="timestamp",
+        name="Observation Timestamp",
```

Comment:
For discussion, not a request: Should the name be "Last Observation Time[stamp]", "Latest Observation Time[stamp]" or something like this?

There was a report from the very beginning of this integration that some stations provide incremental observations, e.g. temperature every 15 minutes but wind speed is only reported in the update every hour (I can't remember the exact details, but something like this).  To solve this problem, pynws uses the most recent data from the requested time interval, and thus in this integration.  But I'm also afraid this distinction is too hard to convey, so also can see the argument for your version of the name.

As an aside, this is one of the reasons why I'm reluctant, but not outright opposed, to let users configure the integration asking for 24 hrs of data from the API, even when this sensor is available.

Reply:
I'm not wedded to any particular name. Out of the ones you suggested, I like "Latest Observation Time," since "Last" sounds a bit final. I only chose the name as a literal concatenation of the API endpoint and key name for consistency.

I know some observations are more granular in time resolution, but I don't think we have to worry about that. If there is a desire to display all of the intervals within the larger timeframe, we can worry about that later (we'd have to modify how `pynws` returns information, anyways right?).

Reply:
> we'd have to modify how `pynws` returns information, anyways right?).

My comment was mostly about naming.  I don't think we should add more complexity here.  I like "Latest Observation Time".

Reply:
This comment was addressed.  For some reason I cannot "resolve" this comment on GitHub.